controllers/service_controller.go

```diff
@@ -110,6 +115,56 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceReconciler) IsOasDefined(ctx context.Context, service *corev1.Service, log logr.Logger) (bool, string, error) {
+	var oasContent string
+	if oasConfigmapName, ok := service.Annotations[KuadrantDiscoveryAnnotationOASConfigMap]; ok {
+		oasContent, err := r.fetchOpenAPIFromConfigMap(
+			ctx,
+			types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace},
+		)
+		log.V(1).Info("get OAS configmap", "objectKey", types.NamespacedName{Name: oasConfigmapName, Namespace: service.Namespace}, "error", err)
+		return true, oasContent, err
+	}
+
+	// The annotations for authodiscovery are the following, if not namedPort, it'll use the first one.
+	// discovery.kuadrant.io/oas-path: "/openapi"
+	// discovery.kuadrant.io/oas-port: openapi
+	if oasURLdefined, ok := service.Annotations[KuadrantDiscoveryAnnotationOASPath]; ok {
+		var targetPort int32 = 80
+		if oasNamePort, ok := service.Annotations[KuadrantDiscoveryAnnotationOASNamePort]; ok {
+			for _, port := range service.Spec.Ports {
+				if port.Name == oasNamePort {
+					targetPort = port.Port
+					// TODO: log messsage here
+					break
+				}
+			}
+		} else {
+			if len(service.Spec.Ports) > 0 {
+				targetPort = service.Spec.Ports[0].Port
+			}
+		}
+
+		// This is terrible hack, we should use the same way as Coredns use it, but
+		// for the moment is ok like this
+		var targetURL = fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", service.Name, service.Namespace, targetPort)
+		resp, err := http.Get(fmt.Sprintf("%s%s", targetURL, oasURLdefined))
+		if err != nil {
+			return true, "", err
+		}
+		if resp.StatusCode != 200 {
+			return true, "", fmt.Errorf("cannot retrieve OAS from %s", targetURL)
```

Comment:
It would be nice to include in the error both the response code and the body content (which generally tries to explain error reason)